diff --git a/src/executors/release-publish/release-publish.impl.js b/src/executors/release-publish/release-publish.impl.js
index 08cd58274d308bcb2ca644e8af5bd99a153c9e91..5ce12c3cb563345ffd19159fd108ff763626b930 100644
--- a/src/executors/release-publish/release-publish.impl.js
+++ b/src/executors/release-publish/release-publish.impl.js
@@ -246,6 +246,8 @@ Please update the local dependency on "${depName}" to be a valid semantic versio
         };
     }
     catch (err) {
+        console.log('Caught Error 1: ') // [macOS]
+        console.log(err) // [macOS]
         try {
             const stdoutData = JSON.parse(err.stdout?.toString() || '{}');
             console.error(`${pm} publish error:`);
@@ -267,6 +269,8 @@ Please update the local dependency on "${depName}" to be a valid semantic versio
             };
         }
         catch (err) {
+            console.log('Caught Error 2: ') // [macOS]
+            console.log(err) // [macOS]
             console.error(`Something unexpected went wrong when processing the ${pm} publish output\n`, err);
             return {
                 success: false,
